src/tools/utils.rs

```diff
@@ -115,3 +119,53 @@ where
     );
     Ok(())
 }
+
+/// block on function
+pub fn block_on<T, F: Future<Output = T>>(f: F) -> Result<T> {
+    Ok(tokio::runtime::Runtime::new()?.block_on(f))
+}
+
+fn index_from_file<K>(
+    header: &IndexHeader,
+    path: &Path,
+) -> AnyResult<BTreeMap<Vec<u8>, Vec<record::Header>>>
+where
+    for<'a> K: Key<'a> + 'static
+{
+    let headers = match header.version() {
+        5 => block_on(async {
+            let index = BPTreeFileIndex::<K>::from_file(
+                FileName::from_path(path)?,
+                None,
+            ).await?;
+            let res = index.get_records_headers(index.blob_size()).await?;
+            AnyResult::<_>::Ok(res.0)
+        })?,
+        _ => return Err(Error::index_header_validation_error("unsupported header version").into()),
+    }?;
+    let headers = headers
+        .into_iter()
+        .map(|(key, value)| (key.to_vec(), value))
+        .collect();
+    Ok(headers)
+}
+
+/// Read index file
+pub fn read_index(path: &Path) -> AnyResult<BTreeMap<Vec<u8>, Vec<record::Header>>> {
```

Comment:
This function should have 2 implementation:
1. Async with all logic
2. Sync, that calls async one with the help of `block_on` defined above